### ISSUE TITLE
IDurableOrchestrationClient.ListInstancesAsync() returns entities along with instances

### ISSUE THREAD

**Issue by yashvardhan90** [Azure/azure-functions-durable-extension#1991]
### Description
> A clear and concise description of what the bug is. Please make an effort to fill in all the sections below; the information will help us investigate your issue.

As per documentation, [IDurableOrchestrationClient.ListInstancesAsync()](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.extensions.durabletask.idurableorchestrationclient.listinstancesasync?view=azure-dotnet) should only return orchestration instances but it is returning entities as well.

### Expected behavior
> A clear and concise description of what you expected to happen.

Return orchestration instances

### Actual behavior
> A clear and concise description of what actually happened.

Returns entities as well as orchestration instances

### App Details

- **Durable Functions extension version (e.g. v1.8.3)**: 2.5.0
- **Azure Functions runtime version (1.0 or 2.0)**: 3.0
- **Programming language used**: C#

### Screenshots
> If applicable, add screenshots to help explain your problem.

### If deployed to Azure
> We have access to a lot of telemetry that can help with investigations. Please provide as much of the following information as you can to help us investigate!

- **Timeframe issue observed**: 10/29/2021
- **Function App name**: DMF-OrchestrationDeployment-dev
- **Function name(s)**: LogManagerOrchestrationStatus
- **Azure region**:
- **Orchestration instance ID(s)**:
- **Azure storage account name**: dmflogsubscriptiondev

> If you don't want to share your Function App or storage account name GitHub, please at least share the orchestration instance ID. Otherwise it's extremely difficult to look up information.

**Comment by sebastianburckhardt** [comment]
I agree this seems like a bug that makes the user experience needlessly complicated (for example, see #1992).

Personally I would love to fix this ASAP, but technically it is a breaking change for anyone who relies on this behavior.

**Comment by sebastianburckhardt** [comment]
It looks to me that this is not the only API call where there is some ambiguity about the distinction between orchestrations and entities. For example, it is not clear that `PurgeInstanceHistoryAsync(DateTime createdTimeFrom, DateTime? createdTimeTo, IEnumerable<OrchestrationStatus> runtimeStatus)` should purge entities.
